src/concatAll.js

```diff
@@ -0,0 +1,34 @@
+import { concat, identical, identity, pipe, when, reduce } from 'ramda';
+
+import stubNull from './stubNull';
+import fl from './fantasy-land/mapping';
+
+const nullSemigroup = { [fl.concat]: identity };
+
+/**
+ * Returns the result of concatenating all semigroups (such as arrays or strings) in passed foldable (such as an array).
+ * Returns null if empty foldable was passed.
+ *
+ * @func concatAll
+ * @memberOf RA
+ * @since {@link https://char0n.github.io/ramda-adjunct/2.6.0|v2.6.0}
+ * @category List
+ * @sig Semigroup s => Foldable s f => f -> (s | null)
+ * @param {Foldable<Semigroup>} foldable foldable with semigroups to concatenate
```

Comment:
`Foldable<Semigroup>` is not true for all version of ramda we are supporting.
I'd go for the following just like [R.concat](http://ramdajs.com/docs/#concat) do
```js
@sig [a] -> [a] -> [a]
@sig String -> String -> String
```

And add a sentence: `Can also concatenate multiple members of a fantasy-land compatible semigroup.`

Reply:
Partly done, added the signatures, but didn't remove the previous one.